### ISSUE TITLE
Swap blank? and blank_as?

### ISSUE THREAD

**Issue by pschambacher** [SamSaffron/fast_blank#15]
Hi!

Any specific reason to have blank? not 100% compatible with ActiveSupport and have a method with a different name which *is* 100% compatible?
I guess that's because the current implementation of blank? is the fastest one but that's also more risky.

Would it be easier for everyone to have blank? 100% compatible and define another fast_blank? method which ignores UTF8 characters?

**Comment by SamSaffron** [comment]
yeah I decided to do so cause I found it odd that the AS version is inconsistent with .empty? I think a consistent implementation of blank is more correct as a default.

**Comment by johnnyshields** [comment]
:+1: +1 for complete parity w/ ActiveSupport `blank?` implementation (as someone who uses a lot of Unicode stuff in Japan--damn you, `\u3000`)

**Comment by SamSaffron** [comment]
I think we should track this here: 

https://bugs.ruby-lang.org/issues/12306

I want to keep parity with `strip.length == 0` out-of-the-box

**Comment by PikachuEXE** [comment]
Is there a way to use this without changing all `blank?` to `blank_as?`
```ruby
# These `require` statements are required?
require "active_support"
require "fast_blank"

# use `blank_as?` as `blank?`
class String
  alias_method :blank?, :blank_as?
end
```

**Comment by SamSaffron** [comment]
not really, but I feel pretty strongly about having parity with
`#strip.empty?` there is talk on ruby about fixing strip so it is unicode
aware, if that happens I will certainly get rid of the old `blank?` and
just have the `blank_as?` version.

**Comment by cantino** [comment]
Until Ruby fixes this, is the recommended way what @PikachuEXE did?

**Comment by SamSaffron** [comment]
yeah pretty much

**Comment by johnnyshields** [comment]
In this gem, we could solve the problem by doing:

```
if defined?(ActiveSupport)
  alias_method :blank?, :blank_as?
end
```

in the future if Ruby supports it we could change this line to:

```
if RUBY_VERSION >= 'x.x.x' || defined?(ActiveSupport)
  alias_method :blank?, :blank_as?
end
```

**Comment by PikachuEXE** [comment]
I guess that can be done with an initializer or something...

**Comment by johnnyshields** [comment]
It can but there's no harm making it part of the gem...

**Comment by semaperepelitsa** [comment]
In case anyone wants to follow, the Ruby core ticket is: https://bugs.ruby-lang.org/issues/12306
